### ISSUE TITLE
push/pull: missing sshfs dependency on macOS

### ISSUE THREAD

**Issue by thias42** [iterative/dvc#6629]
# Bug Report

## Description

Running dvc push returns an error after upgrading to dvc version 2.7.3.:

`
ERROR: failed to push data to the cloud - URL 'ssh://' is supported but requires these missing dependencies: ['sshfs']. Please report this bug to <https://github.com/iterative/dvc/issues>. Thank you!
`

### Reproduce

1. mkdir test
2. cd test
3. git init .
4. dvc init
5. dvc remote add storage ssh://host:/storage
6. dvc remote default storage
7. echo hi > test.txt
8. dvc add test.txt
9. dvc push test.txt

### Expected

I expect the file to be pushed to the remote storage.

### Environment information

macOS Big Sur version 11.5.1 running on MacBook Pro Intel
DVC installed via Homebrew 3.2.12-18-g2dd3934

**Output of `dvc doctor`:**

DVC version: 2.7.3 (brew)
---------------------------------
Platform: Python 3.9.7 on macOS-11.5.2-x86_64-i386-64bit
Supports:
	azure (adlfs = 2021.9.1, knack = 0.8.2, azure-identity = 1.6.1),
	gdrive (pydrive2 = 1.9.3),
	gs (gcsfs = 2021.8.1),
	http (aiohttp = 3.7.4.post0, aiohttp-retry = 2.4.5),
	https (aiohttp = 3.7.4.post0, aiohttp-retry = 2.4.5),
	s3 (s3fs = 2021.8.1, boto3 = 1.17.106),
	webdav (webdav4 = 0.9.1),
	webdavs (webdav4 = 0.9.1)
Cache types: reflink, hardlink, symlink
Cache directory: apfs on /dev/disk1s5s1
Caches: local
Remotes: ssh
Workspace directory: apfs on /dev/disk1s5s1
Repo: dvc, git

**Comment by karajan1001** [comment]
Could you please try
```bash
pip install "dvc[ssh]"
```
or  
```bash
conda install -c conda-forge dvc-ssh
```
to fix the dependency missing?

**Comment by efiop** [comment]
@karajan1001 it is installed from brew, so those won't work 🙁 Looks like we are missing sshfs in https://github.com/Homebrew/homebrew-core/blob/ef2759a348526a7a799eab76dc82c9b2c58c0c1c/pypi_formula_mappings.json#L78

**Comment by efiop** [comment]
Oops, closed by accident. Created a fix in https://github.com/Homebrew/homebrew-core/pull/85313

**Comment by efiop** [comment]
@thias42 Thank you for reporting this issue, we really appreciate it! 🙏 Now just waiting for the PR to get merged by homebrew guys, should be a matter of hours after that to get a new version of the package deployed.

**Comment by thias42** [comment]
Thank you! It is working now again.

**Comment by efiop** [comment]
@thias42 And thank you for the feedback! 🙏

**Comment by TranThanh96** [comment]
Hi, I am using DVC on Linux, and still get that error ? any new updates ?

**Comment by efiop** [comment]
@TranThanh96 Could you provide more info, please? Error log, `dvc doctor` output, etc?

**Comment by TranThanh96** [comment]
![image](https://user-images.githubusercontent.com/26323599/172142633-986512a0-91b0-4392-a96f-a40cf7966510.png)
![image](https://user-images.githubusercontent.com/26323599/172142691-920926c6-d6fa-47c3-a176-8bed37e4ff32.png)

@efiop here

**Comment by efiop** [comment]
@TranThanh96 Could you try installing a newer version of dvc?

Also, please check that `pip freeze` shows both dvc and `sshfs` in the output. In case you have some venv issue. Overall looks like there is some env issue in your setup.

**Comment by TranThanh96** [comment]
> 

dvc==2.10.2
sshfs==2022.3.1

pip freeze show me that version of dvc and sshfs. About env: This is a new env that I created in anaconda.

**Comment by surya-narayanan** [comment]
still facing this issue- same status as @TranThanh96. Are there any solutions / known errors in the env that you suggest looking into @efiop?

**Comment by d-walkama** [comment]
I am having a similar issue on mac:

``
(base) gns-ml-00035:PC_Tempus_DVC dwalkama$ dvc doctor
DVC version: 2.34.2 (conda)
---------------------------------
Platform: Python 3.9.13 on macOS-12.4-x86_64-i386-64bit
Subprojects:
	dvc_data = 0.25.3
	dvc_objects = 0.12.2
	dvc_render = 0.0.12
	dvc_task = 0.1.5
	dvclive = 1.0.1
	scmrepo = 0.1.3
Supports:
	http (aiohttp = 3.8.3, aiohttp-retry = 2.8.3),
	https (aiohttp = 3.8.3, aiohttp-retry = 2.8.3)
Cache types: reflink, hardlink, symlink
Cache directory: apfs on /dev/disk1s5s1
Caches: local
Remotes: local, local, ssh
Workspace directory: apfs on /dev/disk1s5s1
Repo: dvc, git
(base) gns-ml-00035:PC_Tempus_DVC dwalkama$ dvc pull -r refscluster
ERROR: URL 'ssh://' is supported but requires these missing dependencies: ['sshfs']. To install dvc with those dependencies, run:

	conda install -c conda-forge dvc-ssh

See <https://dvc.org/doc/install> for more info.
(base) gns-ml-00035:PC_Tempus_DVC dwalkama$ conda list | grep sshfs
sshfs                     2022.6.0           pyhd8ed1ab_1    conda-forge
``

**Comment by mfakaehler** [comment]
Hi there,

I get the same error in a ubuntu 22.04 installation with miniconda, after trying dvc pull on a ssh remote. This is certainly related to conda and how conda handles these packages. A peculiarity that I observe is:
a) dvc pull in conda base-environment does not work 
b) dvc pull in freshly created test-environment works [^1]
c) a fresh install of miniconda does not resolve the issue

Thus, despite having dvc-ssh and sshfs installed in my conda base-environment I can't have support for ssh. 

Below some diagnostic output. 

Thank you for your support!

The command `dvc pull --verbose` produces:
```
2022-11-24 14:01:57,139 ERROR: URL 'ssh://' is supported but requires these missing dependencies: ['sshfs']. To install dvc with those dependencies, run:

        conda install -c conda-forge dvc-ssh

See <https://dvc.org/doc/install> for more info.
------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc/cli/__init__.py", line 185, in main
    ret = cmd.do_run()
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc/cli/command.py", line 22, in do_run
    return self.run()
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc/commands/data_sync.py", line 31, in run
    stats = self.repo.pull(
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc/repo/__init__.py", line 48, in wrapper
    return f(repo, *args, **kwargs)
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc/repo/pull.py", line 34, in pull
    processed_files_count = self.fetch(
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc/repo/__init__.py", line 48, in wrapper
    return f(repo, *args, **kwargs)
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc/repo/fetch.py", line 68, in fetch
    _remote = self.cloud.get_remote(name=remote)
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc/data_cloud.py", line 66, in get_remote
    fs = cls(**config)
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc_objects/fs/base.py", line 74, in __init__
    self._check_requires(**kwargs)
  File "/opt/miniconda3/lib/python3.9/site-packages/dvc_objects/fs/base.py", line 157, in _check_requires
    raise RemoteMissingDepsError(self, proto, url, missing)
dvc_objects.fs.errors.RemoteMissingDepsError: filesystem for 'ssh': '<class 'dvc_ssh.SSHFileSystem'>' missing dependencies
```
[^1]:  created with `conda create -n dvc-test -c conda-forge dvc dvc-ssh`

Here is `dvc doctor` for the conda base environment:
```
DVC version: 2.34.2 (conda)
---------------------------------
Platform: Python 3.9.15 on Linux-5.15.0-52-generic-x86_64-with-glibc2.35
Subprojects:
        dvc_data = 0.25.3
        dvc_objects = 0.12.2
        dvc_render = 0.0.12
        dvc_task = 0.1.5
        dvclive = 1.0.1
        scmrepo = 0.1.3
Supports:
        http (aiohttp = 3.8.3, aiohttp-retry = 2.8.3),
        https (aiohttp = 3.8.3, aiohttp-retry = 2.8.3)
Cache types:
Cache directory: ext4 on /dev/mapper/3600a09803831435a575d5061456f4f65-part1
Caches: local
Remotes: ssh
Workspace directory: ext4 on /dev/mapper/3600a09803831435a575d5061456f4f65-part1
Repo: dvc, git
```
Here is the output of `conda list | grep 'dvc\|ssh' ` in my base-env, which clearly states that the necessary packages are installed.
```
asyncssh                  2.12.0             pyhd8ed1ab_0    conda-forge
dvc                       2.34.2           py39hf3d152e_0    conda-forge
dvc-data                  0.25.3             pyhd8ed1ab_0    conda-forge
dvc-http                  2.27.2             pyhd8ed1ab_0    conda-forge
dvc-objects               0.12.2             pyhd8ed1ab_0    conda-forge
dvc-render                0.0.12             pyhd8ed1ab_0    conda-forge
dvc-ssh                   2.20.0             pyhd8ed1ab_0    conda-forge
dvc-task                  0.1.5              pyhd8ed1ab_0    conda-forge
dvclive                   1.0.1              pyhd8ed1ab_0    conda-forge
libssh2                   1.10.0               hf14f497_3    conda-forge
sshfs                     2022.6.0           pyhd8ed1ab_1    conda-forge
```

**Comment by efiop** [comment]
@d-walkama Looks like you might've been missing `dvc-ssh` package on conda.

**Comment by efiop** [comment]
@mfakaehler Thanks for the info! Hm, that indeed looks strange. Could you try importing dvc_ssh and sshfs in that environment? Wondering if there is some other import error that is triggering that error.

**Comment by mfakaehler** [comment]
Thank you for your reply @efiop. Forgive my ignorance, what exactly to you mean by importing dvc_ssh and sshfs? They are already installed in the conda-base environment. What should I do, in addition, to import the packages?

**Comment by efiop** [comment]
@mfakaehler I mean to try to import dvc_ssh in that conda environment to see if it produces any errors.

**Comment by mfakaehler** [comment]
I see. Sorry for being so lame... You mean importing dvc_ssh as python module in python? 
`import dvc_ssh` works fine
`import sshfs` indeed causes import errors with openssl and libffi. 
I think this must be related to my conda install. Perhaps due to the fact that the base-env and miniconda install is owned by root. I will look into this and share my progress here. Thanks for pointing me in that direction.

**Comment by efiop** [comment]
@mfakaehler No worries, you are not lame, I just didn't clarify enough 🙂

So looks like those import errors are what is causing dvc to think that sshfs is not installed. Resolving that should fix the problem.

**Comment by mfakaehler** [comment]
I finally resolved the issue by downgrading openssl to 1.1.1s and cffi to 1.15.0 in my base-environment. In addition, I changed owner-ship of the base-environment and miniconda-installation to a non-root user. 
Now sshfs can be properly imported and dvc-ssh works fine.
